gpu/gpu_linux.go

```diff
@@ -70,20 +73,28 @@ func GetCPUMem() (memInfo, error) {
 			_, err = fmt.Sscanf(line, "Buffers:%d", &buffers)
 		case strings.HasPrefix(line, "Cached:"):
 			_, err = fmt.Sscanf(line, "Cached:%d", &cached)
+		case useSwap && strings.HasPrefix(line, "SwapFree:"):
+			_, err = fmt.Sscanf(line, "SwapFree:%d", &freeswap)
+		case useSwap && strings.HasPrefix(line, "SwapTotal:"):
+			_, err = fmt.Sscanf(line, "SwapTotal:%d", &totalswap)
 		default:
 			continue
 		}
 		if err != nil {
 			return mem, err
 		}
 
-		if total > 0 && available > 0 {
+		if total > 0 && available > 0 && useSwap && totalswap > 0 {
+			mem.TotalMemory = (total + totalswap) * format.KibiByte
+			mem.FreeMemory = (available + freeswap) * format.KibiByte
+			return mem, nil
+		} else if total > 0 && available > 0 && !useSwap {
 			mem.TotalMemory = total * format.KibiByte
 			mem.FreeMemory = available * format.KibiByte
 			return mem, nil
 		}
```

Comment:
Unnecessarily complex.  Number variables default to zero, so you can do just:
```suggestion
		if total > 0 && available > 0 {
			mem.TotalMemory = (total + totalswap) * format.KibiByte
			mem.FreeMemory = (available + freeswap) * format.KibiByte
			return mem, nil
		}
```

Reply:
`		if total > 0 && available > 0 {
			mem.TotalMemory = (total + totalswap) * format.KibiByte
			mem.FreeMemory = (available + freeswap) * format.KibiByte
			return mem, nil
		}`
		
		
		This will forcefully add swap.
		If useSwap is enabled then only we can consider adding it